kitsune/messages/api.py

```diff
@@ -0,0 +1,57 @@
+from django.conf import settings
+
+from django.contrib.auth.models import Group, User
+from django.db.models import Q
+from django.views.decorators.http import require_GET
+
+from kitsune.access.decorators import login_required
+from kitsune.sumo.decorators import json_view
+from kitsune.sumo.utils import webpack_static
+from kitsune.users.templatetags.jinja_helpers import profile_avatar
+
+
+@login_required
+@require_GET
+@json_view
+def get_autocomplete_suggestions(request):
+    """An API to provide auto-complete data for user names or groups."""
+    pre = request.GET.get("term", "") or request.GET.get("query", "")
+    if not pre or not request.user.is_authenticated:
+        return []
+
+    def get_suggestions(pre, is_staff=False):
+        """Generate autocomplete suggestions for users and, if applicable, groups."""
+
+        def create_suggestion(item, item_type, is_user=True):
+            """Create a dictionary object for the autocomplete suggestion."""
+            return {
+                "type": item_type,
+                "type_icon": webpack_static(
+                    settings.DEFAULT_USER_ICON if is_user else settings.DEFAULT_GROUP_ICON
+                ),
+                "name": item.username if is_user else item.name,
+                "display_name": item.profile.name if is_user else item.name,
+                "avatar": profile_avatar(item, 24)
+                if is_user
+                else webpack_static(settings.DEFAULT_AVATAR),
+            }
+
+        suggestions = []
+        user_criteria = Q(username__istartswith=pre) | Q(profile__name__istartswith=pre)
+        users = (
+            User.objects.filter(user_criteria, is_active=True)
+            .exclude(profile__is_fxa_migrated=False)
```

Comment:
I checked, and there are 189 active users in production that do not have a profile. Although 189 out of over 4.2 million is very small, perhaps this query could change to ensure that we're not including any users without a profile. I think using `.exclude(profile__is_fxa_migrated=False)` won't exclude users without a profile, but I think using `.filter(user_criteria, is_active=True, profile__is_fxa_migrated=True)` will.

Reply:
Agreed, changed.